src/xviewer-image.c

```diff
@@ -760,8 +757,18 @@ xviewer_image_get_file_info (XviewerImage *img,
                 }
             }
 
+            mime_type_by_extn = g_strdup (g_file_info_get_content_type (file_info));
+
             if (file_type_unknown)
-                *mime_type = g_strdup (g_file_info_get_content_type (file_info));
+               *mime_type = mime_type_by_extn;
```

Comment:
shouldn't this be `g_strdup (mime_type_by_extn)`?

You free it below in 771, and then also back in `xviewer_image_real_load`.

Reply:
Oops, you're correct - it should be with the strdup() call as it is in (what was line 768).

I'll change this and add the spaces after the function names then upload the changed file.